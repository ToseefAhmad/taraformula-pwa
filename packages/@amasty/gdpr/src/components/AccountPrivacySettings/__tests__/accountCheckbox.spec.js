import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import AccountCheckbox from '../accountCheckbox';

jest.mock(
  '@magento/venia-ui/lib/components/Checkbox/checkbox',
  () => 'Checkbox'
);
jest.mock('react-intl', () => ({
  useIntl: jest.fn().mockReturnValue({
    formatMessage: jest
      .fn()
      .mockImplementation(options => options.defaultMessage)
  })
}));
const props = {
  index: 0
};

test('renders no checkbox', () => {
  const tree = createTestInstance(
    <AccountCheckbox
      item={{ hasCheckbox: false, checkboxText: 'test' }}
      {...props}
    />
  );
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders versionChanged false', () => {
  const tree = createTestInstance(
    <AccountCheckbox
      item={{ hasCheckbox: true, checkboxText: 'test' }}
      {...props}
    />
  );
  expect(tree.toJSON()).toMatchSnapshot();
});
