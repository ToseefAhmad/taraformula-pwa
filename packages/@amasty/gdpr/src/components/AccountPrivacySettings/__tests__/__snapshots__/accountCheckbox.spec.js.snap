// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders no checkbox 1`] = `null`;

exports[`renders versionChanged false 1`] = `
<div>
  <Checkbox
    field="privacy.checkbox0"
    label="test"
    validate={[Function]}
    validateOnBlur={true}
  />
</div>
`;
